src/gp_engine.hpp

```diff
@@ -168,15 +205,31 @@ class GPEngine {
   // Rescaling threshold in log space.
   const double log_rescaling_threshold_;
 
-  // ** Per-Node Data
+  // ** Data Sizing
+  // "Count" is the currently occupied by data.
+  // "Padding" is the amount of free working space added to end of occupied space.
+  // "Alloc" is the total current memory allocation.
+  // "Resizing factor" is the amount of extra storage allocated for when resizing.
 
-  // Total number of PLVs across entire DAG. Proportional to the number of nodes in DAG.
-  // plv_count_ = node_count_without_root * plv_per_node.
+  // Total number of nodes in DAG.
+  size_t node_count_ = 0;
+  size_t node_alloc_ = 0;
+  const static size_t node_padding_ = 2;
+  // PLV count proportion to the node_count.
   const size_t plv_count_per_node_ = 6;
-  size_t plv_count_;
+  constexpr static double plv_resizing_factor_ = 2.0;
+  // Total number of edges in DAG. Determines sizes of data vectors stored on edges like
+  // branch lengths.
+  size_t gpcsp_count_ = 0;
+  size_t gpcsp_alloc_ = 0;
+  const static size_t gpcsp_padding_ = 3;
+  constexpr static double gpcsp_resizing_factor_ = 1.0;
```

Comment:
Perhaps you wanted to just have this for symmetry, but I'm not sure if I understand why this is a thing.

Reply:
Yes, it was mostly for symmetry.  I will remove.